features/step_definitions/plans/application_plans_steps.rb

```diff
@@ -131,3 +131,32 @@
   assert_content "The plan was deleted", wait: 10
   step(%(I should not see plan "#{name}"))
 end
+
+And "an application plan that is not default" do
+  @plan = FactoryBot.create(:application_plan, name: "Magic", issuer: @provider.default_service)
+  assert_not @plan.master?
+end
+
+Given "the application plan is published" do
+  @plan.publish!
+  assert @plan.published?
```

Comment:
I would in fact check the published/hidden condition to make the steps more robust:
```ruby
@plan.publish! unless @plan.published?
assert @plan.reload.published?
```